src/aiida/cmdline/commands/cmd_group.py

```diff
@@ -145,41 +145,161 @@ def group_move_nodes(source_group, target_group, force, nodes, all_entries):
 
 
 @verdi_group.command('delete')
-@arguments.GROUP()
+@arguments.GROUPS()
+@options.ALL_USERS(help='Filter and delete groups for all users, rather than only for the current user.')
+@options.USER(help='Add a filter to delete groups belonging to a specific user.')
+@options.TYPE_STRING(default='core', help='Filter to only include groups of this type string.')
+@options.PAST_DAYS(help='Add a filter to delete only groups created in the past N days.', default=None)
+@click.option(
+    '-s',
+    '--startswith',
+    type=click.STRING,
+    default=None,
+    help='Add a filter to delete only groups for which the label begins with STRING.',
+)
+@click.option(
+    '-e',
+    '--endswith',
+    type=click.STRING,
+    default=None,
+    help='Add a filter to delete only groups for which the label ends with STRING.',
+)
+@click.option(
+    '-c',
+    '--contains',
+    type=click.STRING,
+    default=None,
+    help='Add a filter to delete only groups for which the label contains STRING.',
+)
+@options.NODE(help='Delete only the groups that contain a node.')
 @options.FORCE()
 @click.option(
     '--delete-nodes', is_flag=True, default=False, help='Delete all nodes in the group along with the group itself.'
 )
 @options.graph_traversal_rules(GraphTraversalRules.DELETE.value)
 @options.DRY_RUN()
 @with_dbenv()
-def group_delete(group, delete_nodes, dry_run, force, **traversal_rules):
-    """Delete a group and (optionally) the nodes it contains."""
+def group_delete(
+    groups,
+    delete_nodes,
+    dry_run,
+    force,
+    all_users,
+    user,
+    type_string,
+    past_days,
+    startswith,
+    endswith,
+    contains,
+    node,
+    **traversal_rules,
+):
+    """Delete groups and (optionally) the nodes it contains."""
+    from tabulate import tabulate
+
     from aiida import orm
     from aiida.tools import delete_group_nodes
 
-    if not (force or dry_run):
-        click.confirm(f'Are you sure you want to delete {group}?', abort=True)
-    elif dry_run:
-        echo.echo_report(f'Would have deleted {group}.')
+    if groups and (
+        all_users or user or past_days or startswith or endswith or contains or node or type_string != 'core'
```

Comment:
The downside of moving `core` as the default to the argument is that you now no longer know if it was specified by the user or from the default. For example `verdi group delete 123 -T core` now works just fine. This can be misleading for the user, so I think it may be best to remove the default from the option and just assign it in the command body

Reply:
done